Replot resolution data in ISIS calibration when instrument changed

Harrietbrown opened the pull request:
**Description of work.**

This PR fixes a bug in ISISCalibration that causes the resolution data to not be updated when the instrument is changed.
It also correctly sets the defaults for the background and peak ranges when the instrument is changed.

**To test:**

1. open Indirect Data Reduction
2. Go to ISISCalibration
3. in run numbers enter `59057-59059`
4. click run there should be no error
5. Change reflection to 004
6. check that the resolution plot has been updated and the defaults are cahnged to the correct range.
7. click run there should be no error.

re: #33872

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

Harrietbrown committed:
Replot resolution data in ISIS calibration when instrument changed

This commit fixes a bug in ISISCalibration that causes the resolution data to not be updated when the instrument is changed.

re: #33872

Harrietbrown committed:
Change double conversion to int

re: #33872

AnthonyLim23 commented on qt/scientific_interfaces/Indirect/ISISCalibration.cpp:
```diff
@@ -366,8 +366,17 @@ void ISISCalibration::setDefaultInstDetails(QMap<QString, QString> const &instru
 
-  QFileInfo fi(m_lastCalPlotFilename);
+  QString filename = m_uiForm.leRunNo->getFirstFilename();
+  if (filename.isEmpty())
```
how do I trigger this bit of code? I cant seem to think of a way.

Harrietbrown replied:
When the interface is first loaded it should trigger that with `leRunNo` empty

AnthonyLim23 replied:
how do I get line 374 to enter the if?

Harrietbrown committed:
remove unnecessary check that workspace exists

AnthonyLim23 commented:
When I load the initial runs the start values need adjusting (they dont on release).

AnthonyLim23 commented:
The issue seems to be that the default value for peak min is 0 instead of 2900

AnthonyLim23 commented:
As you can see from the screen shot the dashed lines are correct, but the `Peak Min` is 0.0
![image](https://user-images.githubusercontent.com/25006392/169976854-b0c2fbff-0a7f-4f8a-aef9-7edc202bb3b2.png)

Harrietbrown committed:
Disconect range selector signals before setting default range

this commit disables the range selector signals in isiscalibration before the default instrument resolution is set.

re: #33872

AnthonyLim23 commented on qt/scientific_interfaces/Indirect/ISISCalibration.cpp:
```diff
@@ -366,21 +396,24 @@ void ISISCalibration::setDefaultInstDetails(QMap<QString, QString> const &instru
-      setPeakRange(getValueOr(ranges, "peak-start-tof", 0.0), getValueOr(ranges, "peak-end-tof", 0.0));
-      setBackgroundRange(getValueOr(ranges, "back-start-tof", 0.0), getValueOr(ranges, "back-end-tof", 0.0));
-    }
+  disconnect(m_dblManager, SIGNAL(valueChanged(QtProperty *, double)), this, SLOT(calUpdateRS(QtProperty *, double)));
```
why do we need to disconnect and reconnect the signals/slots? Could we just block the signals (something like this https://www.gnu.org/software/libc/manual/html_node/Blocking-for-Handler.html )

Harrietbrown replied:
We disconect because otherwise when we set the default properties first it tries to set the range selector in responce which fails because they are first both set to 0.0, so when we set the min first and it's greater than 0.0 it sets the min to 0.0 which is what you found, by blocking the signal we can set both independantly. I've not seen us use a blocker like that before. if there an example of it being used in mantid?

AnthonyLim23 replied:
I dont know. Will have a look later

AnthonyLim23 commented:
we have blocks via ui file: https://github.com/mantidproject/mantid/blob/main/qt/scientific_interfaces/General/DataComparison.cpp#L104

there is a block in the API https://github.com/mantidproject/mantid/blob/main/qt/scientific_interfaces/Indirect/IndirectPlotOptionsView.cpp#L10

Harrietbrown committed:
Fix incorrect disconnect target.

AnthonyLim23 commented:
It is working much better now. One of the line markers is not matching the value (do you have any ideas why this might be happening?). All of the table values are above 30000, but one of the markers is not.
 
![image](https://user-images.githubusercontent.com/25006392/170451216-f4c6a96e-c063-4d04-8568-43363dc909c4.png)

Harrietbrown committed:
Use set Range instead of set min and max in indirect Tabs

re: #33872

AnthonyLim23 reviewed (approved):
The lines update as expected, there are no crashes and everything runs with the defaults. Looks good to me.
